server/route.go

```diff
@@ -515,7 +515,8 @@ func (c *client) processRouteInfo(info *Info) {
 	if info.Cluster != "" && info.Cluster != clusterName {
 		c.mu.Unlock()
 		// If we are dynamic we may update our cluster name.
-		if s.isClusterNameDynamic() && strings.Compare(clusterName, info.Cluster) < 0 {
+		// Use other if remote is non dynamic or their name is "bigger"
+		if s.isClusterNameDynamic() && (!info.Dynamic || (strings.Compare(clusterName, info.Cluster) < 0)) {
```

Comment:
By the way, I had this comment in the original PR: https://github.com/nats-io/nats-server/pull/1476#discussion_r439693453 and it was not addressed. @derekcollison What you had addressed is the use of isClusterNameDynamic() instead of accessing the cluster name form options here, but not the fact that we should take into account if the remote is dynamic or not.
Without this change, test https://github.com/nats-io/nats-server/pull/1545/files#diff-30fc7b8a07520e58a86a61dbe200433aR1270 fails 100% of the times.